packages/swapper/src/swappers/ThorchainSwapper/endpoints.ts

```diff
@@ -47,7 +48,7 @@ const deductOutboundRuneFee = (fee: string): string => {
 
 export const thorchainApi: SwapperApi = {
   getTradeQuote: async (
-    input: GetTradeQuoteInput,
+    input: GetTradeQuoteInput | GetTradeRateInput,
```

Comment:
As it stands, `GetTradeQuoteInput` also contains all of the trade rate types.

Reply:
Not exactly, one has specific fields explicitly set to undefined which forces us to be explicit in checks vs. adding `| undefined` in base type

Reply:
Correct, I see the distinction between the types, but `GetTradeQuoteInput` is comprised of a union of rate and quote for each type (utxo, evm, cosmossdk) already.

example:
```
export type GetUtxoTradeQuoteInput = GetUtxoTradeQuoteWithWallet | GetUtxoTradeRateInput

export type GetTradeQuoteInput =
  | GetUtxoTradeQuoteInput
  | GetEvmTradeQuoteInput
  | GetCosmosSdkTradeQuoteInput
```

problem is we are now double levaraging functions so naming has become a bit weird. ideally it would be something like `getTradeRateOrQuote(input: GetTradeQuoteInput | GetTradeRateInput)` maybe depending on how we finish building this out. And then `GetTradeQuoteInput` would only contain quote args and `GetTradeRateInput` would only contain rate args.

Reply:
Oh indeed, that union was a smell! https://github.com/shapeshift/web/pull/7939/commits/ee753576f085713474088c70b057ff7c67f10352